File: vol1/ch1/LeetCode26-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int k=0;
        for (int i=1;i<nums.size();i++) {			//首先保留nums[0]
            if (nums[i]==nums[i-1]) k++;      //nums[i]为要删除的元素
            else nums[i-k]=nums[i];         //nums[i]为要保留的元素
        }
        return nums.size()-k;
    }
};

File: vol1/ch1/LeetCode26-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k=0
        for i in range(1,len(nums)):			#首先保留nums[0]
            if nums[i]==nums[i-1]:k+=1      #nums[i]为要删除的元素
            else: nums[i-k]=nums[i]         #nums[i]为要保留的元素
        return len(nums)-k
